homeassistant/components/aws/image_processing.py

```diff
@@ -0,0 +1,483 @@
+                    self,
+                    image,
+                    self.name,
+                    objects,
+                    self.conf_save_file_folder,
+                    self.conf_save_file_timestamp,
+                )
+            await self.async_process_objects(objects, labels)
+
+
+class RekognitionFaceEntity(ImageProcessingFaceEntity):
+    """Rekognition Face Detection/Identification capabilities."""
+
+    _service = "rekognition"
+
+    def __init__(
+        self, camera_entity, session, session_config, entity_config, name=None,
+    ):
+        """Initialize the Rekognition service."""
+        super().__init__()
+        self.session = session
+        self.session_config = session_config
+        self.camera = camera_entity
+        self.collection_id = entity_config[CONF_COLLECTION_ID]
+        self.identify_faces = entity_config.get(CONF_IDENTIFY_FACES, False)
+        self.conf_save_file_folder = entity_config.get(CONF_SAVE_FILE_FOLDER, None)
+        self.conf_save_file_timestamp = entity_config.get(
+            CONF_SAVE_FILE_TIMESTAMP, False
+        )
+        self._confidence = entity_config.get(CONF_CONFIDENCE, 0.0)
+        self.detection_attributes = entity_config.get(
+            CONF_DETECTION_ATTRIBUTES, "DEFAULT"
+        )
+        self.valid_collection_id = False
+        if name:
+            self._name = name
+        else:
+            self._name = f"Rekognition Face {split_entity_id(camera_entity)[1]}"
+
+    @property
+    def confidence(self):
+        """Return minimum confidence for send events."""
+        return self._confidence
+
+    @property
+    def camera_entity(self):
+        """Return camera entity id from process pictures."""
+        return self.camera
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    async def init_collection(self, client):
+        """Create AWS image collection if doesn't exist."""
+        collection_id_req = await client.list_collections(MaxResults=99)
+        if self.collection_id not in collection_id_req["CollectionIds"]:
+            await client.create_collection(CollectionId=self.collection_id)
+        self.valid_collection_id = True
+
+    async def compute_faces(self, client, image):
+        """Detect faces in image and parse response."""
+        index_faces = await client.index_faces(
+            CollectionId=self.collection_id,
+            Image={"Bytes": image},
+            ExternalImageId=self.camera,
+            DetectionAttributes=[self.detection_attributes],
+        )
+        known_faces = []
+        detected_faces = index_faces.get("FaceRecords", [])
+        for face_record in detected_faces:
+            known_face = {}
+            box = await compute_box(face_record["Face"]["BoundingBox"], 3)
+            known_face[ATTR_NAME] = "Unknown"
```

Comment:
from const.py `STATE_UNKNOWN = "unknown"` so lowercase preferred?

Reply:
Now pulling this value directly from `const.py`